### PR TITLE
#23118 Fail SourceQueue completion future in postStop

### LINKED ISSUE #23118: QueueSource does not complete onComplete future on abrupt termination
I haven't verified this with a test but from reading the sources I see that there is no completion of the `onComplete` promise in `postStop` which is the only callback guaranteed to happen if the materializer is shut down while a stream is still running.

### PR DESCRIPTION
**Pull request by 2m**
Fixes #23118

### CODE REVIEW

**Review by jrudolph — APPROVED** [review]
LGTM

File: akka-stream/src/main/scala/akka/stream/impl/QueueSource.scala [hunk 1]
```diff
@@ -53,9 +53,13 @@ import scala.util.control.NonFatal
         if (maxBuffer > 0) buffer = Buffer(maxBuffer, materializer)
         initCallback(callback.invoke)
       }
-      override def postStop(): Unit = stopCallback {
-        case Offer(elem, promise) ⇒ promise.failure(new IllegalStateException("Stream is terminated. SourceQueue is detached"))
-        case _                    ⇒ // ignore
+      override def postStop(): Unit = {
+        val illegalStateException = new IllegalStateException("Stream is terminated. SourceQueue is detached")
```
Comment (johanandren) [inline akka-stream/src/main/scala/akka/stream/impl/QueueSource.scala]: Should it maybe rather be the `AbruptStageTerminationException` which the other stages use in `postStop`?

**Review by johanandren — APPROVED** [review]
LGTM, with one question about exception type

**Review by ktoso — APPROVED** [review]
LGTM, not side about Johan's comment there, either work for me I think

**Comment by johanandren** [comment]
Excellent, thanks Martynas!

**Pull request merged**